pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -171,13 +173,16 @@ func (r *ReconcileAPIServer) Reconcile(request reconcile.Request) (reconcile.Res
 		return reconcile.Result{}, err
 	}
 
-	aci, err := utils.GetAmazonCloudIntegration(ctx, r.client)
-	if errors.IsNotFound(err) {
-		aci = nil
-	} else if err != nil {
-		log.Error(err, "Error reading AmazonCloudIntegration")
-		r.status.SetDegraded("Error reading AmazonCloudIntegration", err.Error())
-		return reconcile.Result{}, err
+	var aci *operatorv1beta1.AmazonCloudIntegration
+	if r.provider == operatorv1.ProviderEKS {
```

Comment:
IMO this is the wrong way around.

I think it should be an error for the users to create an AmazonCloudIntegration resource if the provider is not EKS. IIUC, the current code will just never query the "ACI" unless the provider is EKS.

I think we should query both, and perform cross validation with an error message if an invalid configuration is provided. That way, we can alert the user to incorrect configuration rather than silently ignore it.

Reply:
Hm, I suppose we do this validation here right now: https://github.com/tigera/operator/pull/622/files#diff-f99c8b3f1d4c6a1c8cc66cb40b40e023R162